kubeflow/trainer/backends/container/backend.py

```diff
@@ -0,0 +1,716 @@
+
+        want_all = step == constants.NODE + "-0"
+        for container in sorted(containers, key=lambda c: c["name"]):
+            container_step = container["labels"].get(f"{self.label_prefix}/step", "")
+            if not want_all and container_step != step:
+                continue
+            try:
+                yield from self._adapter.container_logs(container["id"], follow)
+            except Exception as e:
+                logger.warning(f"Failed to get logs for {container['name']}: {e}")
+                yield f"Error getting logs: {e}\n"
+
+    def wait_for_job_status(
+        self,
+        name: str,
+        status: set[str] = {constants.TRAINJOB_COMPLETE},
+        timeout: int = 600,
+        polling_interval: int = 2,
+    ) -> types.TrainJob:
+        import time
+
+        end = time.time() + timeout
+        while time.time() < end:
+            tj = self.get_job(name)
+            logger.debug(f"TrainJob {name}, status {tj.status}")
+            if tj.status in status:
+                return tj
+            if constants.TRAINJOB_FAILED not in status and tj.status == constants.TRAINJOB_FAILED:
+                raise RuntimeError(f"TrainJob {name} is Failed")
+            time.sleep(polling_interval)
+        raise TimeoutError(f"Timeout waiting for TrainJob {name} to reach status: {status}")
+
+    def delete_job(self, name: str):
+        """Delete a training job by querying container runtime."""
+        # Find containers for this job
+        filters = {"label": [f"{self.label_prefix}/trainjob-name={name}"]}
+        containers = self._adapter.list_containers(filters=filters)
+
+        if not containers:
+            raise ValueError(f"No TrainJob with name {name}")
+
+        # Get network_id and workdir from labels
+        network_id = containers[0]["labels"].get(f"{self.label_prefix}/network-id")
+
+        # Get workdir from network labels
+        workdir_host = None
+        if network_id:
+            network_info = self._adapter.get_network(network_id)
+            if network_info:
+                network_labels = network_info.get("labels", {})
+                workdir_host = network_labels.get(f"{self.label_prefix}/workdir")
+
+        # Stop containers and remove
+        from contextlib import suppress
+
+        for container in containers:
+            with suppress(Exception):
+                self._adapter.stop_container(container["id"], timeout=10)
+            with suppress(Exception):
+                self._adapter.remove_container(container["id"], force=True)
+
+        # Remove network (best-effort)
+        if network_id:
+            with suppress(Exception):
+                self._adapter.delete_network(network_id)
+
+        # Remove working directory if configured
+        if self.cfg.auto_remove and workdir_host and os.path.isdir(workdir_host):
+            shutil.rmtree(workdir_host, ignore_errors=True)
+
+    # Helper methods
```

Comment:
Shall we move those helper functions to `backends/container/utils.py` to make it consistent with other directories?